datatypes/mcs_decimal.h

```diff
@@ -598,6 +569,35 @@ class VDecimal: public TSInt128
         return ret;
     }
 
+    inline TSInt128 toTSInt128() const
+    {
+        return TSInt128(s128Value);
+    }
+
+    inline TFloat128 toTFloat128() const
+    {
+        return TFloat128(s128Value);
+    }
+
+    inline double toDouble() const
+    {
+        int128_t scaleDivisor;
+        getScaleDivisor(scaleDivisor, scale);
+        datatypes::TFloat128 tmpval((__float128) s128Value / scaleDivisor);
+        return static_cast<double>(tmpval);
+    }
+
+    inline operator double() const
+    {
+        return toDouble();
+    }
+
+    inline long double toLongDouble() const
+    {
+        datatypes::TFloat128 y(s128Value);
+        return static_cast<long double>(y);
+    }
```

Comment:
The semantics of `toLongDouble()` should be the same as `toDouble()`, i.e., we should use the `scale` here to construct `TFloat128`.

Also, we should also add `operator long double()` as a member function here.

Reply:
To be honest we don't need this function at this point so I'll remove it.